platforms/core-configuration/model-core/src/main/java/org/gradle/internal/extensibility/ConventionAwareHelper.java

```diff
@@ -99,7 +102,10 @@ private MappedProperty map(String propertyName, MappedPropertyImpl mapping) {
     }
 
     private boolean mapConventionOn(SupportsConvention target, MappedPropertyImpl mapping) {
-        if (target instanceof Property) {
+        if (target instanceof DirectoryProperty || target instanceof RegularFileProperty) {
+            ConventionMappingFileSystemLocationPropertyProxy proxy = Cast.uncheckedNonnullCast(target);
+            proxy.conventionFromAny(new DefaultProvider<>(() -> mapping.getValue(_convention, _source)));
```

Comment:
❓ Why wrap this in a `Provider`? I think a `Callable` is perfectly fine, no?

Reply:
This is beacuse convention on property doesn't support Callable (a missing feature?)